types/block.go

```diff
@@ -259,6 +264,128 @@ func mustPush(rowTree *nmt.NamespacedMerkleTree, id namespace.ID, data []byte) {
 	}
 }
 
+func (b *Block) PutBlock(ctx context.Context, api ipfsapi.CoreAPI) error {
```

Comment:
Should we pass in a NodeAdder instead of the whole API directly? 

It looks like we can even just pass in a pinning NodeAdder: https://github.com/ipfs/interface-go-ipfs-core/blob/b935dfe5375eac7ea3c65b14b3f9a0242861d0b3/dag.go#L12

Reply:
Nice! I didn't notice the `Pinning` method in the interface, that simplifies things a fair bit. https://github.com/lazyledger/lazyledger-core/pull/178/commits/29244a00d74f48457456e13163344e742332e90f

Reply:
nice!

Reply:
referencing https://github.com/lazyledger/lazyledger-core/issues/167 s.t. we know that we are already pinning some of the data.